compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -894,7 +896,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
 
         // If we have `rustc_do_not_const_check`, do not check `[const]` bounds.
-        if self.tcx.has_attr(self.body_id, sym::rustc_do_not_const_check) {
+        if self.has_rustc_attrs && self.tcx.has_attr(self.body_id, sym::rustc_do_not_const_check) {
```

Comment:
I'm not certain that this change is correct. OTOH, do we still need the shortcut when the trait solver and const-checking have this check?

Reply:
Oh yes, `self.body_id` is necessarily local, so checking `has_rustc_attrs` is ok.